streamer/input_configuration.py

```diff
@@ -315,10 +315,41 @@ def get_input_args(self) -> List[str]:
   def get_resolution(self) -> bitrate_configuration.VideoResolution:
     return bitrate_configuration.VideoResolution.get_value(self.resolution)
 
+class SinglePeriod(configuration.Base):
+  """An object repersenting one optional video stream and multiple audio and text streams"""
+
+  inputs = configuration.Field(List[Input], required=True).cast()
 
 class InputConfig(configuration.Base):
   """An object representing the entire input config to Shaka Streamer."""
 
-  inputs = configuration.Field(List[Input], required=True).cast()
-  """A list of Input objects, one per input stream."""
+  multiperiod_inputs_list = configuration.Field(List[SinglePeriod]).cast()
+  """A list of SinglePeriod objects"""
+
+  inputs = configuration.Field(List[Input]).cast()
+  """A list of Input objects"""
+
+  def __init__(self, dictionary: Dict[str, Any]):
+    """A constructor to check that either inputs or mutliperiod_inputs_list is provided,
+    and produce a helpful error message in case both or none are provided.
+    
+    We need these checks before passing the input dictionary to the configuration.Base constructor,
+    because it does not check for this 'exclusive or-ing' relationship between fields.
+    """
+
+    assert isinstance(dictionary, dict), """Malformed Input Config File,
+    See some examples at https://github.com/google/shaka-streamer/tree/master/config_files.
+    """
+
+    if dictionary.get('inputs') and dictionary.get('multiperiod_inputs_list'):
+      raise configuration.ConflictingFields(
+        InputConfig, 'inputs', 'multiperiod_inputs_list')
+    
+    # Because these fields are not marked as required at the class level
+    # , we need to check ourselves that one of them is provided.
+    if dictionary.get('inputs') is None and dictionary.get('multiperiod_inputs_list') is None:
```

Comment:
This uses "is None", so empty lists will not be matched.  This means this condition would allow:

```yaml
inputs: []
```

I would prefer to treat this as missing fields, so let's drop use "is None" here and just use truthiness.

Combined with the above suggestion, we would require exactly one field, and that the exactly one field must be non-empty.

We should add tests for these various cases, too.

Reply:
I did want such malformed field error or other types of errors to be caught in the `Base` class constructor instead, that would provide more descriptive error messages about the cause of the error.
But just found out that empty lists between all other types indeed pass this error checking.

Can we add some check in `Base._check_and_convert_type`  that if a list is passed by the user, it must not be empty ??

Reply:
I tried adding this error for the length of the user-inputted list and ran the tests, some `resolution` fields were given as an empty list, after putting some dummy resolutions inside them, the tests passed.

What do you think about this approach?

Reply:
I think you're right.  We should check for empty lists generally.  Possibly empty strings, too.  But let's put that in a separate PR.

I'm thinking of a property on Field that indicates whether empty is acceptable, defaulting to False.